test/MusicStore.Test/ManageControllerTest.cs

```diff
@@ -0,0 +1,116 @@
+
+            // IHttpContextAccessor is required for SignInManager, and UserManager
+            services.AddInstance<IHttpContextAccessor>(new TestHttpContextAcccessor());
+
+            _serviceProvider = services.BuildServiceProvider();
+        }
+
+        [Fact]
+        public async Task Index_ReturnsViewBagMessagesExpected()
+        {
+            // Arrange
+            var userId = "TestUserA";
+            var phone = "abcdefg";
+            var claims = new List<Claim> { new Claim(ClaimTypes.NameIdentifier, userId) };
+
+            var userManager = _serviceProvider.GetRequiredService<UserManager<ApplicationUser>>();
+            var userManagerResult = await userManager.CreateAsync(
+                new ApplicationUser { Id = userId, UserName = "Test", TwoFactorEnabled = true, PhoneNumber = phone },
+                "Pass@word1");
+            Assert.True(userManagerResult.Succeeded);
+
+            var signInManager = _serviceProvider.GetRequiredService<SignInManager<ApplicationUser>>();
+
+            var httpContext = _serviceProvider.GetRequiredService<IHttpContextAccessor>().Value;
+            httpContext.User = new ClaimsPrincipal(new ClaimsIdentity(claims));
+
+            var controller = new ManageController()
+            {
+                ActionContext = new ActionContext(
+                    httpContext,
+                    new RouteData(),
+                    new ActionDescriptor()),
+                UserManager = userManager,
+                SignInManager = signInManager,
+            };
+
+            // Act
+            var result = await controller.Index();
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.NotNull(viewResult.ViewData);
+
+            var model = Assert.IsType<IndexViewModel>(viewResult.ViewData.Model);
+            Assert.True(model.TwoFactor);
+            Assert.Equal(phone, model.PhoneNumber);
+            Assert.True(model.HasPassword);
+        }
+
+        private class TestHttpContextAcccessor : IHttpContextAccessor
+        {
+            private HttpContext _httpContext;
+
+            public TestHttpContextAcccessor()
+            {
+                _httpContext = new TestHttpContext();
+            }
+
+            public HttpContext Value
+            {
+                get
+                {
+                    return _httpContext;
+                }
+
+                set
+                {
+                    _httpContext = value;
+                }
+            }
+        }
+
+        private class TestHttpContext : DefaultHttpContext
+        {
+            public override async Task<IEnumerable<AuthenticationResult>>
+                AuthenticateAsync(IEnumerable<string> authenticationTypes)
+            {
+                return await Task.Run<IEnumerable<AuthenticationResult>>(
+                    () =>
+                     new AuthenticateContext(authenticationTypes).Results);
```

Comment:
use `Task.FromResult` (and drop the `async`) when you need to return sync data from a Task api.